File: code-samples/drivers/ruby/sample/sample.rb
#!/usr/bin/env ruby
require 'cloudcms'

cloudcms = Cloudcms::Cloudcms.new()
platform = cloudcms.connect()
puts "connected to api server at: #{cloudcms.config['baseURL']}"
# puts 'platform: ' + JSON.pretty_generate(platform.data)
puts "platform id: #{platform.data['_doc']}"

repositories = platform.list_repositories
# puts 'list_repositories: ' + JSON.pretty_generate(repositories)

repository = platform.read_repository(repositories[0].data["_doc"])
# puts 'read_repository: ' + JSON.pretty_generate(repository.data)

branches = repository.list_branches
# puts 'branches: ' + JSON.pretty_generate(branches[0].data)
branch = repository.read_branch(branches[0].data["_doc"])

nodes = branch.query_nodes({_type: 'n:node'}, {limit: 5})
node = branch.read_node(nodes[0]["_doc"])
puts 'node: ' + JSON.pretty_generate(node.data)

File: code-samples/drivers/python3/sample/sample.py
from cloudcms import CloudCMS
 
# Connect to Cloud CMS
client = CloudCMS()
platform = client.connect(filename="gitana.json")
 
# List repositories
repositories = platform.list_repositories()
 
# Read repository
repository = platform.read_repository('<repository_id>')
 
# List branches
branches = repository.list_branches()
 
# Read branch
branch = repository.read_branch('<branch_id>')
 
# Read Node
node = branch.read_node('<node_id>')
 
# Create node
obj = {
    'title': 'Twelfth Night',
    'description': 'An old play'
}
newNode = branch.create_node(obj)
 
# Query nodes
query = {
    '_type': 'store:book'
}
pagination = {
    'limit': 2
}
queried_nodes = branch.query_nodes(query, pagination)
 
# Search/Find nodes
find = {
    'search': 'Shakespeare',
    'query': {
        '_type': 'store:book'
    }
}
searched_nodes = branch.find_nodes(find)